worker/processors/hooks/index.js

```diff
@@ -0,0 +1,163 @@
+        const variables = await getHookVariables(hookToExec.id);
+        hookToExec.variables = variables;
+        await executeHook(eventData, hookToExec, eventData.server.id);
+      }
+    }
+  }
+}
+
+
+
+async function executeHook(eventData, hookToExec, serverId) {
+  let server = await SdtdServer.findOne(serverId);
+  eventData.server = server;
+  eventData = await enrichData(eventData);
+  eventData.custom = getVariablesValues(hookToExec.variables, eventData.msg);
+  let results = await sails.helpers.sdtd.executeCustomCmd(server, hookToExec.commandsToExecute.split(';'), eventData);
+  await saveResultsToRedis(hookToExec.id, results);
+  sails.log.debug(`Executed a custom hook for server ${serverId}`, {
+    hook: hookToExec,
+    event: eventData,
+    results: results
+  });
+}
+
+async function executeLogLineHook(eventData, hookToExec, serverId) {
+
+  // Ignore log messages created by the web API. This is the prevent spam
+  if (eventData.msg.includes('WebCommandResult_for_')) {
+    return;
+  }
+
+  let server = await SdtdServer.findOne(serverId);
+  eventData.server = server;
+  eventData = await enrichData(eventData);
+  eventData.custom = getVariablesValues(hookToExec.variables, eventData.msg);
+  let results = await sails.helpers.sdtd.executeCustomCmd(server, hookToExec.commandsToExecute.split(';'), eventData);
+  await saveResultsToRedis(hookToExec.id, results);
+  sails.log.debug(`Executed a custom logLine hook for server ${serverId}`, {
+    hook: hookToExec,
+    event: eventData,
+    results: results
+  });
+}
+
+
+// Checks if the logline matches the searchString or regex
+function checkLogLine(logLine, hook) {
+  let logLineMatchesSearch = true;
+
+  if (!_.isEmpty(hook.searchString)) {
+    logLineMatchesSearch = logLine.includes(hook.searchString);
+  }
+
+  if (!_.isEmpty(hook.regex)) {
+    logLineMatchesSearch = (new RegExp(hook.regex)).test(logLine);
+  }
+
+  return logLineMatchesSearch;
+}
+
+/**
+*
+* @param {Object} hook The hook that is being executed
+* @returns {boolean} true: okay to execute, false: hook is still on cooldown
+*/
+async function handleCooldown(hook) {
+  if (hook.cooldown) {
+    let lastExecutionTime = await sails.helpers.redis.get(`hooks:${hook.id}:lastExecutionTime`);
+
+    if (_.isNull(lastExecutionTime)) {
+      lastExecutionTime = 0;
+    }
+
+    lastExecutionTime = parseInt(lastExecutionTime);
+
+    let currentTime = Date.now();
+
+    if (lastExecutionTime + hook.cooldown > currentTime) {
+      return false;
+    } else {
+      await sails.helpers.redis.set(`hooks:${hook.id}:lastExecutionTime`, currentTime);
+      return true;
+    }
+  } else {
+    return true;
+  }
+}
+
+
+async function getHookVariables(hookId) {
+  let variables = await HookVariable.find({
+    hook: hookId
+  });
+  return variables;
+}
+
+function getVariablesValues(variables, logMsg) {
+  const customVars = {};
+
+  for (const variable of variables) {
+    const regex = new RegExp(variable.regex);
```

Comment:
I was afraid that new regexp was like an eval, but I can't find anything that says that
But that being said, we should use https://www.npmjs.com/package/safe-regex since (a+){10} is apparently bad and potentially easy to typo.